include/nuttx/mutex.h

```diff
@@ -42,6 +44,13 @@
 
 typedef sem_t mutex_t;
 
+typedef struct
+{
+  mutex_t mutex;
+  pid_t holder;
+  uint16_t count;
+} rmutex_t;
```

Comment:
```suggestion
struct rmutex_s
{
  mutex_t mutex;
  pid_t holder;
  uint16_t count;
};

typedef struct rmutex_s rmutex_t;
```
NuttX coding standard forbids unnamed structs:
https://nuttx.apache.org/docs/latest/contributing/coding_style.html#structures